Update Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs

GrzegorzDryja opened the pull request:
Update made thanks to PM suggestions.
1. I actually thought that this function could be used for the entire project
2. I think I did it
3. At the moment, no matter, it probably disappeared
4. Ok thanks
5. No problem
6. I meant the future GUI interface ;)

Commit:
Update Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs

Update made with PM sugestions.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -6,25 +6,23 @@ namespace Notatnik_parser.Notepad
-        string note;
-
-        public void SaveFile()
+        private string SaveFile()
```
now, this methos is responsible for two functionalities: get text from console and save it to file. Try create private method for get text and add string parameter to SaveFile function. Then you can use them in Run()

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -6,25 +6,23 @@ namespace Notatnik_parser.Notepad
-            ConsoleKeyInfo saveButton = Console.ReadKey();
-            if (saveButton.Key == ConsoleKey.Enter)
+            
+            ConsoleKeyInfo forSave = Console.ReadKey();
```
I think the name is still not clear. What "forSave" tells you? What is forSave here? Name should tell you exacly function of the varible. Maybe somthing like "typedKey"? Try find different name ;)

GrzegorzDryja replied:
Oh, mail does not goes where I supposed.

tomku-mar reviewed (changes requested):
Naming variables and functions is very important and hard. I have problems with it myself

Commit:
Added SaveFile function, new variable name

I've made SaveFile function, but couldn't make it work without global variable as You wish at first.

GrzegorzDryja commented:
Ok, now I get it. "typedKey" name for this variable is actually very good
name in this case.

śr., 7 lis 2018 o 17:56 tomkowy <notifications@github.com> napisał(a):

> *@tomkowy* commented on this pull request.
> ------------------------------
>
> In Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs
> <https://github.com/tomkowy/notatnik_parser/pull/25#discussion_r231589816>
> :
>
> >          }
>
>          public void Run()
>          {
>              Console.WriteLine("Witamy w notatniku. Wprowadź notatkę (wciśnij <enter> aby zapisać):");
> -            note = Console.ReadLine();
> -
> -            ConsoleKeyInfo saveButton = Console.ReadKey();
> -            if (saveButton.Key == ConsoleKey.Enter)
> +
> +            ConsoleKeyInfo forSave = Console.ReadKey();
>
> I think the name is still not clear. What "forSave" tells you? What is
> forSave here? Name should tell you exacly function of the varible. Maybe
> somthing like "typedKey"? Try find different name ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tomkowy/notatnik_parser/pull/25#pullrequestreview-172592755>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AdfzY0Nbc46T_hnBHkqTiF2QtRUKZt3bks5usxC3gaJpZM4YPVam>
> .
>

GrzegorzDryja commented:
I know, I've watch few lessons about this and one speaker on Code
Europe emphasized
it strongly. But practice is different. Thank you for looking on code from
your perspective.

śr., 7 lis 2018 o 18:00 tomkowy <notifications@github.com> napisał(a):

> *@tomkowy* requested changes on this pull request.
>
> Naming variables and functions is very important and hard. I have problems
> with it myself
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tomkowy/notatnik_parser/pull/25#pullrequestreview-172594994>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AdfzY-tBqkFZEWU3bOpzIm9B6RVdhuVyks5usxGwgaJpZM4YPVam>
> .
>

Commit:
Passing Parameters

I've made saveFile function static with reference parameter.

Commit:
Return to main menu

Added return to main menu function from program class after saving note.
SaveFile function is not static anymore

Commit:
Update SaveFile function

Global use of SaveFile function was on my mind. Now SaveFile function is set to private.
"ref" is mentioned in official MS guide - https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/passing-parameters
Code works now without ref key word.

Commit:
Merge pull request #26 from tomkowy/issue#24

Return to main menu is now one branch with notepad for easy reading.

GrzegorzDryja commented:
I've made some changes, from that moment please check "notepad" branch for merge.

tomku-mar commented on Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:
```diff
@@ -5,27 +5,35 @@
+
+            Console.WriteLine("Witamy w notatniku. Wprowadź notatkę (wciśnij 2 x <enter> aby zapisać):");
+
+            note = Console.ReadLine();
```
You can assign a variable in one line:
string note = Console.ReadLine();
Youre code is fine, but i think that my proposition is more readable (one line of code less)

GrzegorzDryja replied:
Yes, I know. BTW Most of all I like to use this kind of text format - while curly braces is on the end of the line - it shorten number of lines a lot, [check this commit](https://github.com/tomkowy/notatnik_parser/blob/issue%2322-curly-braces-on-the-end/Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs)
To work with this project I switched to match other files :)

tomku-mar reviewed (approved):
:)

GrzegorzDryja commented:
Finally merging my branch made me very proud of myself!
